test/simple_sliding_window_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+
+class TestSimpleSlidingWindow < MiniTest::Unit::TestCase
+  CLASS = ::Semian::Simple::SlidingWindow
+
+  def setup
+    @sliding_window = CLASS.new(max_size: 6)
+    @sliding_window.clear
+  end
+
+  def teardown
+    @sliding_window.destroy
+  end
+
+  module SlidingWindowTestCases
+    def test_sliding_window_push
+      assert_equal(0, @sliding_window.size)
+      @sliding_window << 1
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 1, 1, 6)
+      @sliding_window << 5
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 5, 2, 6)
+    end
+
+    def test_sliding_window_resize
+      assert_equal(0, @sliding_window.size)
+      @sliding_window << 1 << 2 << 3 << 4 << 5 << 6
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 6, 6, 6)
+      @sliding_window.resize_to 6
+      assert_correct_first_and_last_and_size(@sliding_window, 1, 6, 6, 6)
+      @sliding_window.resize_to 5
+      assert_correct_first_and_last_and_size(@sliding_window, 2, 6, 5, 5)
+      @sliding_window.resize_to 6
+      assert_correct_first_and_last_and_size(@sliding_window, 2, 6, 5, 6)
+    end
+
+    def test_sliding_window_edge_falloff
+      assert_equal(0, @sliding_window.size)
+      @sliding_window << 0 << 1 << 2 << 3 << 4 << 5 << 6 << 7
+      assert_correct_first_and_last_and_size(@sliding_window, 2, 7, 6, 6)
+      @sliding_window.shift
+      assert_correct_first_and_last_and_size(@sliding_window, 3, 7, 5, 6)
+    end
+  end
+
+  module SlidingWindowUtilityMethods
+    def assert_correct_first_and_last_and_size(sliding_window, first, last, size, max_size)
+      assert_equal(first, sliding_window.first)
+      assert_equal(last, sliding_window.last)
+      assert_equal(size, sliding_window.size)
+      assert_equal(max_size, sliding_window.max_size)
+    end
```

Comment:
These assertions are hard to read. Since your window for testing is fairly small, I'd go with something very explicit asserting on the entire window:

```ruby
assert_sliding_window [1]
```

Which is just implemented as an `assert_equal` on the underlying array.

With that, you implicitly assert on `max_size` and `size` which is IMO better and much easier to read.